runtime/onert/core/include/ir/train/ITrainableOperation.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ONERT_IR_ITRAINABLE_OPERATION_H__
+#define __ONERT_IR_ITRAINABLE_OPERATION_H__
+
+#include "ir/Operation.h"
+
+namespace onert
+{
+namespace ir
+{
+
+struct OperationVisitor;
+
+namespace train
+{
+
+struct TrainableOperationVisitor;
+
+class ITrainableOperation : public IOperation
+{
+public:
+  virtual ~ITrainableOperation() = default;
+
+public:
+  virtual void accept(TrainableOperationVisitor &v) const = 0;
```

Comment:
I missing out the following patch.
```suggestion
  virtual void accept(OperationVisitor &v) const override = 0;
  virtual void accept(TrainableOperationVisitor &v) const = 0;
```